src/coreclr/src/jit/emitarm.cpp

```diff
@@ -7676,7 +7676,7 @@ void emitter::emitInsLoadStoreOp(instruction ins, emitAttr attr, regNumber dataR
 {
     GenTree* addr = indir->Addr();
 
-    if (addr->isContained())
+    if (addr->isContained() && !addr->OperIs(GT_PHYSREG))
```

Comment:
Can you explain why you couldn't use the form `IND(LEA(PHYSREG r11))` and then not need these special cases, instead of using `IND(PHYSREG r11)` that requires the special case? In both cases, the physreg would get marked contained, and the LEA would also be marked contained.

Reply:
From what I understand, `PHYSREG` still assigns a new register (in this case `x4`) and not reuse `x11` because it is occupied. Hence we generate extra `mov x4, x11`.

```
                                                              /--*  t16    long   
Generating: N025 (???,???) [000030] ------------        t30 = *  PUTARG_REG long   REG x11
Generating: N027 (  1,  1) [000031] ------------        t31 =    PHYSREG   long   x11 REG x4
IN0007:                           mov     x4, x11
                                                              /--*  t31    long   
Generating: N029 (  2,  2) [000032] -c----------        t32 = *  LEA(b+0)  long   REG NA
                                                              /--*  t32    long   
Generating: N031 (  5,  4) [000033] ------------        t33 = *  IND       long   REG x4
IN0008:                           ldr     x4, [x4]
```